src/lib/DepositApiClient.js

```diff
@@ -111,8 +103,8 @@ export class RDMDepositApiClient extends DepositApiClient {
   async createDraft(draft) {
     const payload = this.recordSerializer.serialize(draft);
     return this._createResponse(() =>
-      axiosWithConfig.post(this.createDraftURL, payload, {
-        headers: BASE_HEADERS["vnd+json"],
+      this.axiosWithConfig.post(this.createDraftURL, payload, {
+        headers: this.apiHeaders["vnd+json"],
```

Comment:
with BASE_HEADERS removed, we cannot assure `vnd+json` to be present in the apiHeaders object

Reply:
The BASE_HEADERS are set in the configuration in the app inveniosoftware/invenio-app-rdm#1917.

Reply:
The BASE_HEADERS will be set in the invenio_app_rdm and allows other modules to set their own headers.

Reply:
I understand, however this solution relies on defining the headers in a repository which depends on react-invenio-deposit. As a result there is a "mutual dependency" created - module cannot be used without invenio-app-rdm. 
If the assumption of having `vnd+json` header in this.apiHeaders object is not fulfiled then headers will be evaluated as `undefined`

Reply:
I see your point and pushed a solution where no mutual dependency is created.